Allow configuration to set start of week

namdnguyen opened the pull request:
Create a week_start variable in the .timetrap.yml configuration file. Maintain the default of Monday as the start of the week when parsing `timetrap week`.

Update help information on `timetrap week` usage.

Pull request for #141.

samg commented:
Looks good.  Could you also add some test coverage to make sure this isn't inadvertently broken in the future.  You can use the `with_stubbed_config` helper to change timetrap's config within the test context.

namdnguyen commented:
Sure thing! I'll work I those to submit.

namdnguyen commented:
I changed the 'week' function to be more like the 'month' one, so that it would be easier to create tests for it. Added a couple of tests for the week_start configuration, too. Let me know if it needs anything else!

namdnguyen commented on lib/timetrap/cli.rb:
```diff
@@ -443,7 +447,12 @@ def yesterday
 
     def week
-      args['-s'] = Date.today.wday == 1 ? Date.today.to_s : Date.parse(Chronic.parse(%q(last monday)).to_s).to_s
+      d = Chronic.parse( args['-s'] || Date.today )
```
Actually, now that I look at this, I should probably remove allowing the `-s` argument here, otherwise it will take everything from the `-s` date until today, which could be more than a week. 

Alternatively, changing how the end date argument in line 455 is set could be more complex, if you want to allow for someone to set a start date for week. Then timetrap would go back to the last day that match `week_start` configuration and show all entries from that date and 6 days following it. The date specified in the user-inputted value for `t week -s` would be contained in this query. This would match the behavior for `t month`, which allows for pulling up previous months.

namdnguyen replied:
I went ahead and submitted a commit that sets end_of_week similarly to how month sets end_of_month.

samg commented:
Sorry for the delay reviewing this. Looks like good functionality to me.  Thanks for adding tests.  I'll push out gem version asap.

samg commented:
https://rubygems.org/gems/timetrap/versions/1.15.0

namdnguyen commented:
Awesome. Thanks for reviewing and accepting the pull request!
